source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -0,0 +1,302 @@
+        for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
+          if (selected_properties[property_index])
+            b[property_index] = 0;
+
+        unsigned int positions_index = 0;
+        const double cell_diameter = found_cell->diameter();
+        for (typename ParticleHandler<dim>::particle_iterator particle = particle_range.begin();
+             particle != particle_range.end(); ++particle, ++positions_index)
+          {
+            const auto &particle_property_value = particle->get_properties();
+            for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
+              if (selected_properties[property_index])
+                b[property_index][positions_index] = particle_property_value[property_index];
+            const Tensor<1, dim, double> relative_particle_position = (particle->get_location() - approximated_cell_midpoint) / cell_diameter;
+            // A is accessed by A[column][row] here since we need to append
+            // columns into the qr matrix.
+
+            // There is a potential that in the future we may need to add more
+            // 'mixed' terms to the interpolation for some problems. We for now
+            // leave those terms out of the interpolation because they would
+            // require more particles per cell and we are not yet aware of
+            // a benchmark where those terms have mattered.
+            A[0][positions_index] = 1;
+            A[1][positions_index] = relative_particle_position[0];
+            A[2][positions_index] = relative_particle_position[1];
+            if (dim == 2)
+              {
+                A[3][positions_index] = relative_particle_position[0] * relative_particle_position[1];
+                A[4][positions_index] = relative_particle_position[0] * relative_particle_position[0];
+                A[5][positions_index] = relative_particle_position[1] * relative_particle_position[1];
+              }
+            else
+              {
+                A[3][positions_index] = relative_particle_position[2];
+                A[4][positions_index] = relative_particle_position[0] * relative_particle_position[1];
+                A[5][positions_index] = relative_particle_position[0] * relative_particle_position[2];
+                A[6][positions_index] = relative_particle_position[1] * relative_particle_position[2];
+                A[7][positions_index] = relative_particle_position[0] * relative_particle_position[0];
+                A[8][positions_index] = relative_particle_position[1] * relative_particle_position[1];
+                A[9][positions_index] = relative_particle_position[2] * relative_particle_position[2];
+              }
+          }
+
+        for (unsigned int column_index = 0; column_index < n_matrix_columns; ++column_index)
+          qr.append_column(A[column_index]);
```

Comment:
This is the first place where you actually use the `qr` variable. Move the declaration of it right above this loop.

Reply:
You could also write this loop in a more elegant way as
```
  for (const auto &column : A)
    qr.append_column(column);
```